### PR TITLE
Initialize the project description for GSoD

### PR DESCRIPTION
**Pull request by oleg-nenashev**
subj. CC @StackScribe

### CODE REVIEW

File: mentorship/gsod/2022/project-ideas.md [hunk 1]
```diff
@@ -5,16 +5,84 @@ which Keptn contributors have proposed for this year.
 Application guidelines and more information about Keptn
 in GSoD are available on [this page](../README.md).
 
-## 2022 Project Idea
+## 2022 Project. Re-architecting and improving Keptn user documentation
 
-### Re-architecting and improving Keptn user documentation
+During GSoC 2022 we will run either one large-size project or two small-size ones.
+You will work with the static Keptn website and the documentation engine based on the modern technology stack.
+Currently we use Hugo for the website and Docs, and Google Codelabs for tutorials.
```
Comment (imsingh) [inline mentorship/gsod/2022/project-ideas.md:12]:
```suggestion
Currently, we use Hugo for the website and Docs, and Google Codelabs for tutorials.
```

File: mentorship/gsod/2022/project-ideas.md [hunk 2]
```diff
@@ -5,16 +5,84 @@ which Keptn contributors have proposed for this year.
...
+We plan adoption Docusaurus or Antora as a new documentation platform during Google Summer of Code 2022
+([project idea](https://github.com/keptn/community/blob/main/mentorship/gsoc/2022/project-ideas.md#new-documentation-site-engine)).
+Technical writers in this project will be able to work with one of these technologies while
+using the Documentation-as-Code approach and Mardown/Asccidoc languages.
```
Comment (imsingh) [inline mentorship/gsod/2022/project-ideas.md:16]:
```suggestion
using the Documentation-as-Code approach and Markdown/Asciidoc languages.
```

File: mentorship/gsod/2022/project-ideas.md [hunk 3]
```diff
@@ -5,16 +5,84 @@ which Keptn contributors have proposed for this year.
...
+
+Create and test installation instructions for creating a k8s cluster on which Keptn can be installed for demo and evaluation purposes.
+Create excellent instructions for installing Keptn CLI and Keptn on this new cluster.
+Confirm that one set of instructions works for all k8s implementations and all “Keptn curious” projects we provide. 
+If necessary, create multiple instruction sets, clearly defining when each is appropriate.
+
+Also, we propose to create and test instructions for installing a Keptn “playground” in an existing k8s cluster for study and demos. 
+The goal here is to provide an environment where someone can try Keptn out with their existing data without risking damage to an existing production system (such as Prometheus, Dynatrace, DataDog, etc.)
+
+For this deliverable, instructions should be pedagogical rather than pure cookbook:
```
Comment (imsingh) [inline mentorship/gsod/2022/project-ideas.md:42]:
```suggestion
For this deliverable, instructions should be pedagogical rather than a pure cookbook:
```

File: mentorship/gsod/2022/project-ideas.md [hunk 4]
```diff
@@ -5,16 +5,84 @@ which Keptn contributors have proposed for this year.
...
+#### New example projects
+
+Create 5-6 projects that introduce key Keptn capabilities for the Keptn curious to explore. 
+We have a number of existing tutorials to use as starters or inspiration. 
+Note that many of the existing tutorials use the [sockshop project](https://github.com/keptn-demo/sockshop) as a base;
+and some updates will be needed there.
+
+Each project should include:
+
+- Files that can be downloaded from github to provide some basic configuration with an explanation of what is provided in these files (that they would have to do themselves if they wanted to implement this functionality.
```
Comment (imsingh) [inline mentorship/gsod/2022/project-ideas.md:55]:
```suggestion
- Files that can be downloaded from GitHub to provide some basic configuration with an explanation of what is provided in these files (that they would have to do themselves if they wanted to implement this functionality.
```

File: mentorship/gsod/2022/project-ideas.md [hunk 5]
```diff
@@ -5,16 +5,84 @@ which Keptn contributors have proposed for this year.
...
+- Specification in GitHub for what needs to be installed to provide a viable environment for the project, including integrations that must be included, configurations that must be set, and so forth.
```
Comment (imsingh) [inline mentorship/gsod/2022/project-ideas.md:56]:
```suggestion
- The specification in GitHub for what needs to be installed to provide a viable environment for the project, including integrations that must be included, configurations that must be set, and so forth.
```

File: mentorship/gsod/2022/project-ideas.md [hunk 6]
```diff
@@ -5,16 +5,84 @@ which Keptn contributors have proposed for this year.
...
+- Excellent information about how to run the sample project and do the exercises, including context for each step, what to look for to determine success and understand the output.  The goal is to begin to educate the user about how to get the most out of Keptn
```
Comment (imsingh) [inline mentorship/gsod/2022/project-ideas.md:57]:
```suggestion
- Excellent information about how to run the sample project and do the exercises, including context for each step, what to look for to determine success, and understanding the output.  The goal is to begin to educate the user about how to get the most out of Keptn
```

File: mentorship/gsod/2022/project-ideas.md [hunk 7]
```diff
@@ -5,16 +5,84 @@ which Keptn contributors have proposed for this year.
...
+
+#### Separate guides for roles
+
+We want to introduce separate documentation for different Keptn user roles: 
+administrators, end users and developers.
```
Comment (imsingh) [inline mentorship/gsod/2022/project-ideas.md:62]:
```suggestion
administrators, end-users, and developers.
```

File: mentorship/gsod/2022/project-ideas.md [hunk 8]
```diff
@@ -5,16 +5,84 @@ which Keptn contributors have proposed for this year.
...
+We suggest to work together to regroup the content so that the new structure is introduced.
+
+#### New reference pages
+
+Rewrite reference pages for files to conform to the structure proposed in the [Structure of reference pages](https://docs.google.com/document/d/1rmSGUFsnYuNc_BGHGmkW-9iB3hrOYbMeA-mJmMOjGRk/edit?usp=sharing) proposal.
+We are currently creating reference pages for critical files by capturing information in existing doc files, source code README files, etc.
+This project would be to restructure and expand the information in these files to create proper reference pages.
+It might also include relocating some information in existing files to a more appropriate location.
+
+Same for other standard documentation independent from version.
```
Comment (imsingh) [inline mentorship/gsod/2022/project-ideas.md:72]:
```suggestion
Same for other standard documentation independent from the version.
```

File: mentorship/gsod/2022/project-ideas.md [hunk 9]
```diff
@@ -5,16 +5,84 @@ which Keptn contributors have proposed for this year.
...
+
+Creating missing guides is not part of this project
 
 ## Alternative project ideas
 
+We also discussed a number of alternative project ideas that can be considered.
+
 ### Documentation assessment
 
 This project would assess the Keptn documentation, to help us come with a plan for improving our documentation. In particular, we're looking for someone to look at the content organization, the clarity of the language and concepts, and to make sure it's as readable as possible for both implementors and end users.
```
Comment (imsingh) [inline mentorship/gsod/2022/project-ideas.md:88]:
```suggestion
This project would assess the Keptn documentation, to help us come up with a plan for improving our documentation. In particular, we're looking for someone to look at the content organization, the clarity of the language and concepts, and to make sure it's as readable as possible for both implementors and end-users.
```

File: mentorship/gsod/2022/project-ideas.md [hunk 10]
```diff
@@ -25,3 +93,16 @@ This project would assess the Keptn documentation, to help us come with a plan f
...
+
+Architect information about how to contribute to Keptn
+(and potentially related projects),
+get community approval for the proposal, then implement it. 
+For some preliminary notes about what is currently there and issues, see
+[this proposal](https://docs.google.com/document/d/10rhdsh3E-d4zYq1jJzS6cJM78x1ViyPvRZVkn-h4HgY/edit?usp=sharing).
+
+### More ideas
+
+For ideas that didn't make to this list,
```
Comment (imsingh) [inline mentorship/gsod/2022/project-ideas.md:107]:
```suggestion
For ideas that didn't make it to this list,
```

**Review by imsingh — CHANGES_REQUESTED** [review]
Some minor suggestions

File: mentorship/gsod/2022/project-ideas.md [hunk 11]
```diff
@@ -5,16 +5,84 @@ which Keptn contributors have proposed for this year.
...
 in GSoD are available on [this page](../README.md).
 
-## 2022 Project Idea
+## 2022 Project. Re-architecting and improving Keptn user documentation
 
-### Re-architecting and improving Keptn user documentation
+During GSoC 2022 we will run either one large-size project or two small-size ones.
+You will work with the static Keptn website and the documentation engine based on the modern technology stack.
+Currently we use Hugo for the website and Docs, and Google Codelabs for tutorials.
+We plan adoption Docusaurus or Antora as a new documentation platform during Google Summer of Code 2022
```
Comment (StackScribe) [inline mentorship/gsod/2022/project-ideas.md:13]:
```suggestion
We plan to adopt either Docusaurus or Antora as a new documentation platform during Google Summer of Code 2022.
```

File: mentorship/gsod/2022/project-ideas.md [hunk 12]
```diff
@@ -5,16 +5,84 @@ which Keptn contributors have proposed for this year.
...
+([project idea](https://github.com/keptn/community/blob/main/mentorship/gsoc/2022/project-ideas.md#new-documentation-site-engine)).
+Technical writers in this project will be able to work with one of these technologies while
+using the Documentation-as-Code approach and Mardown/Asccidoc languages.
 
-> Final idea will be published soon.
-> It will be a combination of ideas documented
-> [here](https://docs.google.com/document/d/154zrnkUdUhyWyDB-Gs123JvbiYPb-zQDLaV2Oe0PC8A/edit#heading=h.ckoz6t2d3tab).
+### Key deliverables
+
+We offer the technical writers to focus on 2-3 deliverables from the list below:
```
Comment (StackScribe) [inline mentorship/gsod/2022/project-ideas.md:20]:
```suggestion
Technical writers should focus on 2-3 deliverables from the list below:
```
I don't know if this is correct but the existing syntax does not work.  Maybe it should be something like "We offer the following list, from which technical writers can choose 2-3 deliverables on which to focus." ?  Or something else.

File: mentorship/gsod/2022/project-ideas.md [hunk 13]
```diff
@@ -5,16 +5,84 @@ which Keptn contributors have proposed for this year.
...
+
+- Files that can be downloaded from github to provide some basic configuration with an explanation of what is provided in these files (that they would have to do themselves if they wanted to implement this functionality.
+- Specification in GitHub for what needs to be installed to provide a viable environment for the project, including integrations that must be included, configurations that must be set, and so forth.
+- Excellent information about how to run the sample project and do the exercises, including context for each step, what to look for to determine success and understand the output.  The goal is to begin to educate the user about how to get the most out of Keptn
+
+#### Separate guides for roles
+
+We want to introduce separate documentation for different Keptn user roles: 
+administrators, end users and developers.
+We suggest to work together to regroup the content so that the new structure is introduced.
```
Comment (StackScribe) [inline mentorship/gsod/2022/project-ideas.md:63]: I like this idea but some people on the mentors mailing list have talked about how important it is that each participant's work is completely independent of the work of all other participants.  Will the GSoC organizers see this as running afoul of that guideline?  What if this said, "Each participant is expected to contribute to the design of the new structure and then implement an assigned portion of content for the new structure" or something?

File: mentorship/gsod/2022/project-ideas.md [hunk 14]
```diff
@@ -5,23 +5,104 @@ which Keptn contributors have proposed for this year.
...
+The goal here is to provide an environment where someone can try Keptn out with their existing data without risking damage to an existing production system (such as Prometheus, Dynatrace, DataDog, etc.)
+
+For this deliverable, instructions should be pedagogical rather than a pure cookbook:
+provide enough explanation that the user gains some understanding of what they are doing.
+Include information for verifying each step and some troubleshooting information or links.
+
+#### New example projects
+
+Create 5-6 projects that introduce key Keptn capabilities for the Keptn curious to explore. 
+We have a number of existing tutorials to use as starters or inspiration.
```
Comment (mehabhalodiya) [inline mentorship/gsod/2022/project-ideas.md:49]:
```suggestion
We have some existing tutorials to use as starters or inspiration. 
```

File: mentorship/gsod/2022/project-ideas.md [hunk 15]
```diff
@@ -5,23 +5,104 @@ which Keptn contributors have proposed for this year.
...
+
+- Files that can be downloaded from GitHub to provide some basic configuration with an explanation of what is provided in these files (that they would have to do themselves if they wanted to implement this functionality.
+- The specification in GitHub for what needs to be installed to provide a viable environment for the project, including integrations that must be included, configurations that must be set, and so forth.
+- Excellent information about how to run the sample project and do the exercises, including context for each step, what to look for to determine success, and understanding the output.  The goal is to begin to educate the user about how to get the most out of Keptn
+
+#### Separate guides for roles
+
+We want to introduce separate documentation for different Keptn user roles: 
+administrators, end-users, and developers.
+We suggest to work together to regroup the content so that the new structure is introduced.
```
Comment (mehabhalodiya) [inline mentorship/gsod/2022/project-ideas.md:63]:
```suggestion
We suggest working together to regroup the content so that the new structure is introduced.
```

File: mentorship/gsod/2022/project-ideas.md [hunk 16]
```diff
@@ -5,23 +5,104 @@ which Keptn contributors have proposed for this year.
...
+located in the version-specific sections of the documentation.
+Then modify the existing Concepts articles to conform to this standard and create Concepts articles for appropriate concepts that
+are discussed in existing guides or elsewhere
+(such as README files in the software repos or existing Keptn videos).
+
+Creating missing guides is not part of this project
 
 ## Alternative project ideas
 
+We also discussed a number of alternative project ideas that can be considered.
```
Comment (mehabhalodiya) [inline mentorship/gsod/2022/project-ideas.md:84]:
```suggestion
We also discussed several alternative project ideas that can be considered.
```

File: mentorship/gsod/2022/project-ideas.md (hunk 8 above)
Comment (StackScribe) [reply mentorship/gsod/2022/project-ideas.md]: Is it "independent from the version" or is it "written to be relevant for all versions"?  The point is to document differences between versions rather than expect the reader to do textural analysis on the docs for different versions to determine when changes occur.

File: mentorship/gsod/2022/project-ideas.md [hunk 17]
```diff
@@ -5,16 +5,84 @@ which Keptn contributors have proposed for this year.
...
+Rewrite reference pages for files to conform to the structure proposed in the [Structure of reference pages](https://docs.google.com/document/d/1rmSGUFsnYuNc_BGHGmkW-9iB3hrOYbMeA-mJmMOjGRk/edit?usp=sharing) proposal.
+We are currently creating reference pages for critical files by capturing information in existing doc files, source code README files, etc.
+This project would be to restructure and expand the information in these files to create proper reference pages.
+It might also include relocating some information in existing files to a more appropriate location.
+
+Same for other standard documentation independent from version.
+Set a standard for what information belongs in a “Concepts” article versus
+what belongs in reference pages or guides that are currently
+located in the version-specific sections of the documentation.
+Then modify the existing Concepts articles to conform to this standard and create Concepts articles for appropriate concepts that
```
Comment (StackScribe) [inline mentorship/gsod/2022/project-ideas.md:76]:
```suggestion
Then modify the existing Concepts articles to conform to this standard and create Concepts articles for appropriate topics that
```

File: mentorship/gsod/2022/project-ideas.md (hunk 13 above)
Comment (afzal442) [reply mentorship/gsod/2022/project-ideas.md:63]: >Each participant is expected to contribute to the design of the new structure and then implement an assigned portion of content for the new structure

Yes, I echo that. As per GSoC guidelines, it has to be contributed individually to the assigned project, but not to work together on the same project.

File: mentorship/gsod/2022/project-ideas.md (hunk 9 above)
Comment (StackScribe) [reply mentorship/gsod/2022/project-ideas.md]: "come up" is not great phraseology for formal writing.  How about "to help us devise a plan for improving our documentation."?  Could be "create a plan" or "define a plan" or something else.

File: mentorship/gsod/2022/project-ideas.md (hunk 10 above)
Comment (StackScribe) [inline mentorship/gsod/2022/project-ideas.md]:
```suggestion
For ideas that were proposed [or discussed] but not included this list,
```

File: mentorship/gsod/2022/project-ideas.md (hunk 14 above)
Comment (mehabhalodiya) [reply mentorship/gsod/2022/project-ideas.md:49]: The phrase `a number of` may be wordy. So, just a little suggestion 🙂

File: mentorship/gsod/2022/project-ideas.md (hunk 16 above)
Comment (StackScribe) [reply mentorship/gsod/2022/project-ideas.md:84]: Could it just be "We also discussed alternative project ideas..." or even just "We discussed other project ideas..."

File: mentorship/gsod/2022/project-ideas.md [hunk 18]
```diff
@@ -5,23 +5,104 @@ which Keptn contributors have proposed for this year.
...
+
+Architect information about how to contribute to Keptn
+(and potentially related projects),
+get community approval for the proposal, then implement it. 
+For some preliminary notes about what is currently there and issues, see
+[this proposal](https://docs.google.com/document/d/10rhdsh3E-d4zYq1jJzS6cJM78x1ViyPvRZVkn-h4HgY/edit?usp=sharing).
+
+### More ideas
+
+For ideas that didn't make it to this list,
```
Comment (StackScribe) [inline mentorship/gsod/2022/project-ideas.md:107]:
```suggestion
For a list of potential projects that are not included in this list,
```

File: mentorship/gsod/2022/project-ideas.md [hunk 19]
```diff
@@ -5,23 +5,104 @@ which Keptn contributors have proposed for this year.
 Application guidelines and more information about Keptn
 in GSoD are available on [this page](../README.md).
 
-## 2022 Project Idea
+## 2022 Project. Re-architecting and improving Keptn user documentation
 
-### Re-architecting and improving Keptn user documentation
+During GSoC 2022 we will run either one large-size project or two small-size ones.
+You will work with the static Keptn website and the documentation engine based on the modern technology stack.
+Currently, we use Hugo for the website and Docs, and Google Codelabs for tutorials.
```
Comment (StackScribe) [inline mentorship/gsod/2022/project-ideas.md:12]:
```suggestion
Currently, we use Hugo to build the website and documentation; and Google Codelabs to build the tutorials.
```

File: mentorship/gsod/2022/project-ideas.md [hunk 20]
```diff
@@ -5,23 +5,104 @@ which Keptn contributors have proposed for this year.
...
+
+Technical writers should focus on 2-3 deliverables from the list below:
+
+- Installation Guidelines for Keptn and Keptn CLI client on common Kubernetes clusters.
+  This deliverable will update or replace the existing documentation.
+- New example projects that demonstrate key Keptn capabilities to newcomer users and potential adopters.
+  This deliverable will update or replace the existing documentation and tutorials.
+- Restructure existing documentation to user and administrator guidelines
+  and developer guides.
+- Reference pages for key information and concepts that are not specific to Keptn versions.
```
Comment (StackScribe) [inline mentorship/gsod/2022/project-ideas.md:28]:
```suggestion
- Restructure existing guides and explanatory documentation into user, administrator,
  and developer guides.
- Reference pages for key information and concepts that are appropriate for all Keptn versions.
```

File: mentorship/gsod/2022/project-ideas.md (hunk 13 above)
Comment (StackScribe) [reply mentorship/gsod/2022/project-ideas.md:63]: It's a delicate subject because, of course, collaboration is so essential to all successful open source projects.  I'm thinking that, by stating that everyone contributes to the plan, it's the plan that may suffer if a participant fails to deliver their chunk of the materail but this would not leave another participant unable to proceed with their work.

**Review by StackScribe — APPROVED** [review]
lgtm.  A few editorial suggestions and other remarks for you can consider.

File: mentorship/gsod/2022/project-ideas.md (hunk 16 above)
Comment (mehabhalodiya) [reply mentorship/gsod/2022/project-ideas.md:84]: All well! This is also good for me :)

File: mentorship/gsod/2022/project-ideas.md (hunk 13 above)
Comment (oleg-nenashev) [reply mentorship/gsod/2022/project-ideas.md]: I suggest rewording in a follow-up pull request. I need to focus on parenting, and I may not be available until the deadline
